src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splittedString = sourceString
+    .split(';')
+    .map((str) => str.split(':'))
+    .map((str) => str.map((value) => value.trim()))
+    .map((str) => str.filter((value) => value !== ''))
+    .filter((str) => str.length !== 0)
```

Comment:
The chained methods have the correct indentation, which is good for readability. However, you can combine the two `map` calls into one to make the code more concise. Instead of mapping twice, you can trim the values and split the string in the same iteration.

Reply:
Agreed with bot. There are a lot of loops which you can remove. Please feel free to ask for help in fe_chat.